naga/src/proc/mod.rs

```diff
@@ -732,6 +739,52 @@ impl GlobalCtx<'_> {
     }
 }
 
+pub enum ResolvedSize {
+    Constant(u32),
+    Runtime,
+}
+
+#[derive(Error, Debug, Clone, Copy, PartialEq)]
+pub enum ResolveArraySizeError {
+    #[error("array element count must be positive (> 0)")]
+    ExpectedPositiveArrayLength,
+}
+
+/// # Panics
```

Comment:
I think the docs should say something like:

```suggestion
/// Return the number of elements that `size` represents, if known at code generation time.
///
/// This must only be called after `back::pipeline_constants::process_overrides`, or on modules
/// that contain no overrides.
///
/// # Panics
```